packages/config/src/lib/index.ts

```diff
@@ -0,0 +1,106 @@
+import { LogType, fecLogger } from '@redhat-cloud-services/frontend-components-config-utilities';
```

Comment:
Github wasn't able to diff this with the old file location so I went ahead and did it locally:
```diff
➜  Desktop diff config-index-old.js config-index-new.ts -w
1,2c1,3
< const config = require('./src/config');
< const plugins = require('./src/plugins');
---
> import { LogType, fecLogger } from '@redhat-cloud-services/frontend-components-config-utilities';
> import createConfig, { CreateConfigOptions } from './createConfig';
> import createPlugins, { CreatePluginsOptions } from './createPlugins';
3a5,6
> export * from './createConfig';
> export * from './createPlugins';
4a8,13
> type DevServerConfiguration = import('webpack-dev-server').Configuration;
> type WebpackConfiguration = import('webpack').Configuration;
> export interface FecWebpackConfiguration extends WebpackConfiguration {
>   devServer?: DevServerConfiguration;
> }
> 
11c20
< const getAppEntry = (rootFolder, isProd) => {
---
> const getAppEntry = (rootFolder: string, isProd?: boolean) => {
16c25
<       console.warn('Found multiple entry-dev files. Using', entries[0]);
---
>       fecLogger(LogType.warn, `Found multiple entry-dev files. Using: ${entries[0]}`);
26c35
<     console.warn('Found multiple entry files. Using', entries[0]);
---
>     fecLogger(LogType.warn, `Found multiple entry files. Using: ${entries[0]}`);
32c41,53
< module.exports = (configurations) => {
---
> type FecConfigurationOptions = Omit<CreateConfigOptions, 'publicPath' | 'appEntry' | 'appName'> &
>   CreatePluginsOptions & {
>     deployment?: string;
>     debug?: boolean;
>     appEntry?: string;
>   };
> 
> const createFecConfig = (
>   configurations: FecConfigurationOptions
> ): {
>   config: ReturnType<typeof createConfig>;
>   plugins: ReturnType<typeof createPlugins>;
> } => {
40c61
<     console.log('[fec] no git branch detected, using main for webpack "main" config.');
---
>     fecLogger(LogType.info, 'no git branch detected, using main for webpack "main" config.');
50,59c71,80
<     /* eslint-disable no-console */
<     console.log('~~~Using variables~~~');
<     console.log(`Root folder: ${configurations.rootFolder}`);
<     console.log(`Current branch: ${gitBranch}`);
<     !generateSourceMaps && console.log(`Source map generation for "${gitBranch}" deployment has been disabled.`);
<     console.log(`Beta branches: ${betaBranches}`);
<     console.log(`Using deployments: ${appDeployment}`);
<     console.log(`Public path: ${publicPath}`);
<     console.log(`App entry: ${appEntry}`);
<     console.log(`Use proxy: ${configurations.useProxy ? 'true' : 'false'}`);
---
>     console.group();
>     fecLogger(LogType.debug, '~~~Using variables~~~');
>     fecLogger(LogType.debug, `Root folder: ${configurations.rootFolder}`);
>     fecLogger(LogType.debug, `Current branch: ${gitBranch}`);
>     !generateSourceMaps && fecLogger(LogType.debug, `Source map generation for "${gitBranch}" deployment has been disabled.`);
>     fecLogger(LogType.debug, `Beta branches: ${betaBranches}`);
>     fecLogger(LogType.debug, `Using deployments: ${appDeployment}`);
>     fecLogger(LogType.debug, `Public path: ${publicPath}`);
>     fecLogger(LogType.debug, `App entry: ${appEntry}`);
>     fecLogger(LogType.debug, `Use proxy: ${configurations.useProxy ? 'true' : 'false'}`);
61,62c82,83
<       console.warn('Insights-proxy is deprecated in favor of "useProxy" or "standalone".');
<       console.warn('See https://github.com/RedHatInsights/frontend-components/blob/master/packages/config/README.md');
---
>       fecLogger(LogType.warn, 'Insights-proxy is deprecated in favor of "useProxy" or "standalone".');
>       fecLogger(LogType.warn, 'See https://github.com/RedHatInsights/frontend-components/blob/master/packages/config/README.md');
65c86
<     console.log('~~~~~~~~~~~~~~~~~~~~~');
---
>     console.groupEnd();
70c91
<     config: config({
---
>     config: createConfig({
72,73d92
<       appDeployment,
<       insights,
78c97
<     plugins: plugins({
---
>     plugins: createPlugins({
81,83c100
<       appDeployment,
<       insights,
<       publicPath,
---
>       appname: insights.appname,
86a104,106
> 
> export default createFecConfig;
> module.exports = createFecConfig;
\ No newline at end of file
```